webui/react/src/App.tsx

```diff
@@ -89,6 +89,20 @@ const AppView: React.FC = () => {
   // Correct the viewport height size when window resize occurs.
   useLayoutEffect(() => correctViewportHeight(), [ resize ]);
 
+  useEffect(() => {
+    checkServerAlive().then(isAlive => {
+      if (!isAlive) {
+        notification.warn({
+          btn: <Button type="primary" onClick={refreshPage}>Retry</Button>,
+          description: `We cannot communicate with the server at "${serverAddress()}".
+          Please check firewall or server settings.`,
+          duration: 0,
+          message: 'Server Unreachable',
+        });
+      }
+    });
+  }, []);
+
```

Comment:
**suggestion:** what do you think about displaying in the center of the `App` view the error message and the refresh button instead of having a notification popup on a blank page?

Reply:
hmm I was thinking of the case that we do the same UX with a heart-beat check so the user could be doing something and the server goes away. In that case it might be nicer to avoid covering the whole screen. but if we're addressing just the current use case sending the user to an offline page might be a better experince

Reply:
added an offline page.